### PR TITLE
fix(router) atc compat escaping

### PR DESCRIPTION
**Pull request by Suika-Kong**
Fix FT-3257

### CODE REVIEW

File: kong/router/atc_compat.lua [hunk 1]
```diff
@@ -132,6 +132,10 @@ function _M._set_ngx(mock_ngx)
 end
 
 
+local function atc_escape_str(str)
+  return " \"" .. str:gsub([[\]], [[\\]]):gsub([["]], [[\"]]) .. "\""
+end
+
```
Comment (chronolaw) [inline kong/router/atc_compat.lua:138]: should be 2 blank lines.

File: kong/router/atc_compat.lua [hunk 2]
```diff
@@ -132,6 +132,10 @@ function _M._set_ngx(mock_ngx)
 end
 
 
+local function atc_escape_str(str)
+  return " \"" .. str:gsub([[\]], [[\\]]):gsub([["]], [[\"]]) .. "\""
```
Comment (chronolaw) [inline kong/router/atc_compat.lua:136]: Should we also consider `'`?

File: spec/01-unit/08-router_spec.lua [hunk 3]
```diff
@@ -3571,6 +3570,66 @@ for _, flavor in ipairs({ "traditional", "traditional_compatible", "expressions"
...
+              protocol = "https"
+            },
+            route      = {
+              id = "e8fb37f1-102d-461e-9c51-6608a6bb8102",
+              paths    = { [[~/\d+"]] },
+            },
+          },
+        }
+
+        local router = assert(Router.new(use_case_routes))
```
Comment (chronolaw) [inline spec/01-unit/08-router_spec.lua:3600]: should use function `new_router` to test flavor `expressions`.

File: kong/router/atc_compat.lua (hunk 2 above)
Comment (Suika-Kong) [reply kong/router/atc_compat.lua:136]: `'` need not be escaped.

File: spec/01-unit/08-router_spec.lua (hunk 3 above)
Comment (Suika-Kong) [reply spec/01-unit/08-router_spec.lua:3600]: I'm just moving this part of code.

File: kong/router/atc_compat.lua (hunk 1 above)
Comment (mayocream) [reply kong/router/atc_compat.lua:138]:
```suggestion


```

**Review by chronolaw — APPROVED** [review]

**Review by mayocream — APPROVED** [review]

**Review by dndx — APPROVED** [review]

**Pull request merged**